packages/flutter_tools/lib/src/ios/device_ios.dart

```diff
@@ -546,52 +546,48 @@ Future<bool> _buildIOSXcodeProject(ApplicationPackage app, bool isDevice) async
+  _copyServiceFrameworks(services, frameworksDirectory);
+
+  // Step 3: Copy the service definitions manifest at the correct spot for
+  //         xcodebuild to pick up.
+  File manifestFile = new File(path.join(bundle.path, "ServiceDefinitions.json"));
+  _copyServiceDefinitionsManifest(services, manifestFile);
 }
 
-Future _fetchFrameworks(List<Map<String, String>> services) async {
+void _copyServiceFrameworks(List<Map<String, String>> services, Directory frameworksDirectory) {
+  printTrace("Copying service frameworks to '${path.absolute(frameworksDirectory.path)}'");
+  frameworksDirectory.createSync(recursive: true);
   for (Map<String, String> service in services) {
-    String frameworkUrl = service['framework'];
-    service['framework-path'] = await getServiceFromUrl(
-       frameworkUrl, service['root'], service['name'], unzip: true);
-  }
-}
-
-void _copyFrameworksToBundle(String destDir, List<Map<String, String>> services) {
-  // TODO(mpcomplete): check timestamps.
-  for (Map<String, String> service in services) {
-    String basename = path.basename(service['framework-path']);
-    String destPath = path.join(destDir, basename);
-    _copyDirRecursive(service['framework-path'], destPath);
+    File dylib = new File(path.join(service['root'], service['ios-framework']));
+    printTrace("Copying ${dylib.path} into bundle");
+    if (!dylib.existsSync()) {
+      printError("The service dylib '${dylib.path}' does not exist");
+      continue;
+    }
+    // Shell out so permissions on the dylib are preserved
+    runCheckedSync(['/bin/cp', dylib.path, frameworksDirectory.path]);
   }
 }
 
-void _copyDirRecursive(String fromPath, String toPath) {
-  Directory fromDir = new Directory(fromPath);
-  if (!fromDir.existsSync())
-    throw new Exception('Source directory "${fromDir.path}" does not exist');
-
-  Directory toDir = new Directory(toPath);
-  if (!toDir.existsSync())
-    toDir.createSync(recursive: true);
-
-  for (FileSystemEntity entity in fromDir.listSync()) {
-    String newPath = '${toDir.path}/${path.basename(entity.path)}';
-    if (entity is File) {
-      entity.copySync(newPath);
-    } else if (entity is Directory) {
-      _copyDirRecursive(entity.path, newPath);
-    } else {
-      throw new Exception('Unsupported file type for recursive copy.');
-    }
-  };
+void _copyServiceDefinitionsManifest(List<Map<String, String>> services, File manifest) {
+  printTrace("Creating service definitions manifest at '${manifest.path}'");
+  List<Map<String, String>> jsonServices = services.map((Map<String, String> service) => {
+    'name': service['name'],
+    // Since we have already moved it to the Frameworks directory. Strip away
+    // the directory and basenames.
+    'framework': path.basenameWithoutExtension(service['ios-framework'])
+  }).toList();
+  Map<String, dynamic> json = { 'services' : jsonServices };
+  manifest.writeAsStringSync(JSON.encode(json), mode: FileMode.WRITE, flush: true);
```

Comment:
I'd prefer this to be moved back to services.json to be shared with the android code. We should also use the same filename on both platforms.